frontend/projects/upgrade/src/app/shared/services/common-form-helpers.service.ts

```diff
@@ -0,0 +1,14 @@
+import { Injectable } from '@angular/core';
+import { FormGroup } from '@angular/forms';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class CommonFormHelpersService {
+  triggerTouchedToDisplayErrors(form: FormGroup) {
```

Comment:
what is this service for?

Reply:
because we'll likely need this exact function for every form, and no other place existed yet. and I'm sure we'll have more form utility helpers as we go along. I've created a "table helper" service as well in a different PR. I want to make sure we have clear and obvious places to put shared utility functions, so it makes sense to me to have common table, form, and dialog services for all those little reusable functions that get copy-pasted between components otherwise.

Reply:
ok, I just want to know the use case.